lib/layout.js

```diff
@@ -24,28 +24,34 @@ function traverse (node, callback) {
 module.exports = function (depth, height, rootOffset, children) {
 
   return function (root) {
-    var stack = [root]
-      , nodes = []
-      , node
+    if (!root) {
+      return []
+    }
 
-    root.depth = 0
+    var roots = Array.isArray(root) ? root : [root]
+      , nodes = []
 
-    while ((node = stack.pop()) != null) {
-      nodes.push(node)
-      if ((childs = children.call(null, node, node.depth)) && (n = childs.length)) {
-        var n, childs, child
-        while (--n >= 0) {
-          stack.push(child = childs[n])
-          child.parent = node
-          child.depth = node.depth + 1
+    roots.forEach(function (_root) {
+      var stack = [_root]
+        , node, n, childs, child
+      _root.depth = 0
+
+      while ((node = stack.pop()) != null) {
+        nodes.push(node)
+        if ((childs = children.call(null, node, node.depth)) && (n = childs.length)) {
```

Comment:
Nope, but you pointed out a flaw in my code. I wanted the children callback to receive the tree object as `this`. Thanks. Will update.